ocs_ci/ocs/ui/base_ui.py

```diff
@@ -255,10 +259,24 @@ def take_screenshot(self):
             self.screenshots_folder,
             f"{datetime.datetime.now().strftime('%Y-%m-%dT%H-%M-%S.%f')}.png",
         )
-        logger.info(f"Creating snapshot: {filename}")
+        logger.info(f"Creating screenshot: {filename}")
         self.driver.save_screenshot(filename)
         time.sleep(0.5)
 
+    def copy_dom(self):
+        """
+        Get page source of the webpage
+
+        """
+        filename = os.path.join(
+            self.screenshots_folder,
+            f"{datetime.datetime.now().strftime('%Y-%m-%dT%H-%M-%S.%f')}_DOM.txt",
+        )
+        logger.info(f"Copy DOM file: {filename}")
+        html = self.driver.page_source
+        with open(filename, "w") as f:
```

Comment:
Shouldn't this be w+ as it also create file if not exist?

Reply:
it is working :)

Reply:
Ok, I thought that it's needed or maybe was needed in older python version.